### DISCUSSION TITLE
One-to-many relationship

### DISCUSSION THREAD

**Discussion by Rizialdi** [prisma/prisma#2236] [Q&A] [answered]
Good morning everyone. I recently decided to switch from *Prisma 1* to *Prisma 2* and after changing the schema.prisma file, i tried to get the same behavior as before. However I can't figure out where I did things wrong. When I try to create a message with the mutation createMessage I get the error:

```sh
Error: Message creation failed
    at createMessage (/.../server/graphQL/resolvers/Message.js:29:17)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

The hint is that it comes from:

```js
const message = await context.prisma.message.create({ data: { text, sentBy: { connect: { id: userId } }, channel: { connect: { id: channelId } } }})
```
but I don't know if its during prisma schema definition or during resolver I get things wrong. 
I provided below the related schema.prisma types:
```js
// /typeDefs/Messages.js
 type Message {
    id: ID!
    text: String!
    sentBy: User!
    channel: Channel!
    createdAt: String!
  }   

type Mutation {
    createMessage(channelId: String, recipient: String, text: String!): Boolean!
  }
```

```prisma
// /prisma/schema.prisma
model Message {
  id         String     @id @default(cuid())
  text       String
  sentById   String?     @map("sentBy")
  sentBy     User?       @relation(fields: [sentById], references: [id])
  channelId  String?     @map("channel")
  channel    Channel?    @relation(fields: [channelId], references: [id])
  createdAt  DateTime   @default(now())
}

model Channel {
  id         String     @id @default(cuid())
  users      User[]     @relation(references: [id])
  messages   Message[]  @relation(references: [id])
  createdAt  DateTime   @default(now())
}

model User {
  id         String     @id @default(cuid())
  nom        String
  avatar     String?
  prenom     String
  numero     String     @unique
  createdAt  DateTime   @default(now())
  updatedAt  DateTime   @updatedAt
  channels   Channel[]
  offerings  Offering[]
  messages   Message[]
}
```
# Environment setup:

- OS: Ubuntu 19.10
- Database: PostgreSQL
- Prisma version: 2.0.0-beta.2
- Node version: v13.13.0

**Comment by Rizialdi** [comment]
Good morning everyone.
Do you have any clue on the issue ?

**Comment by ryands17** [comment, marked as answer]
Hey @Rizialdi :wave: 
You had an issue in your schema, the `@relation` directive is not required on the many side of the relation while using 1-M relations

I have updated the schema here:

```prisma
model Message {
  id        String   @default(cuid()) @id
  text      String
  sentBy    User?    @relation(fields: [sentById], references: [id])
  sentById  String?  @map("sentBy")
  channel   Channel? @relation(fields: [channelId], references: [id])
  channelId String?  @map("channel")
  createdAt DateTime @default(now())
}

model Channel {
  id        String    @default(cuid()) @id
  users     User[]    @relation(references: [id])
  messages  Message[]
  createdAt DateTime  @default(now())
}

model User {
  id        String    @default(cuid()) @id
  nom       String
  avatar    String?
  prenom    String
  numero    String    @unique
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  channels  Channel[]
  messages  Message[]
}
```

Let me know if this works!

You can read more about the new relation syntax [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations)

**Reply by Rizialdi** [reply]
Thanks a lot, it worked like a charm.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
